kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/modem/ModemDriver.java

```diff
@@ -256,17 +289,13 @@ private void toggleGpio(String gpio, String gpioPath, String directionPath, Stri
         }
 	}
 	
-	private void invertGpioValue(String gpio, String gpioPath, String directionPath, String valuePath) throws IOException {       
-        exportGpio(gpio, gpioPath);  //Prepare gpios
-        setGpioDirection(directionPath, "out");
-        
+	private void invertGpioValue(String gpio, String gpioPath, String directionPath, String valuePath) throws IOException {
         FileReader fGpioOldValueReader= null;  //read current value
         BufferedReader fGpioOldValue= null;
         int oldValue= 0;
         try {
             fGpioOldValueReader= new FileReader(valuePath);
-            fGpioOldValue= new BufferedReader(fGpioOldValueReader);
-            oldValue= Integer.parseInt(fGpioOldValue.readLine());
+            oldValue = fGpioOldValueReader.read() - 48;
```

Comment:
@ibinshtok Honestly I prefer the old buffered reader instead of having this line and its magic number